### ISSUE TITLE
Application Error

### ISSUE THREAD

**Issue by ManarKamel** [Rob--W/cors-anywhere#42]
http://cors-anywhere.herokuapp.com/ Application Error:
how long is this going to last ?

**Comment by Rob--W** [comment]
Investigating. The error is "Backlog too deep", which means that the server receives more requests than it can handle.

**Comment by Rob--W** [comment]
Looks like #32 again. More than half of the requests are to plusone.google.com.

**Comment by Rob--W** [comment]
I am getting 250 requests per second...

**Comment by Rob--W** [comment]
Out of the 15k requests that were received, half of them fail because the load is too high (codes: https://devcenter.heroku.com/articles/error-codes)

```
$ grep -oP 'code=\KH\d+' /tmp/oneminute | sort -n | uniq -c
   7710 H11
   3991 H12
      1 H19
    741 H27
```

These are the status codes that are sent to clients.
```
$ grep -oP 'status=\K\d+' /tmp/oneminute | sort -n | uniq -c
   1269 200
     15 400
   1265 403
    741 499
  11702 503
```

Ignoring the failing requests, these are the status codes:

```
$ grep -vP 'code=\KH\d+' /tmp/oneminute  | grep -oP 'status=\K\d+' | sort -n | uniq -c
   1269 200
     15 400
   1265 403
```

The majority (1250+) of these 403 errors are the result of blocking the requests from #32. Then there is one destination site that is used for the majority of the other requests (1100+), with the rest using not even a dozen requests. Someone is abusing this public proxy :(

**Comment by ManarKamel** [comment]
this has been going on for awhile, Heroku servers are not that good, you should put some limits for requests or create some virtual queue or something.

Check this CORS proxy: https://crossorigin.me/
Github: https://github.com/technoboy10/crossorigin.me

Which is based on https://github.com/bmpvieira/simple-corsproxy

this CORS proxy is pretty good same as yours!

**Comment by Rob--W** [comment]
This issue is not caused by the server logic. The server can handle the requests just fine if Heroku passed the request down to the application and sufficient (network) resources are available. However, due to the huge number of pending requests the backlog grows (up to a point where Heroku starts discarding requests). Any server can be brought on its knees if you use up all resources.

I've analyzed the problem, and found that the responsible site (from my previous comment) does not need to proxy requests, it proxies requests through CORS Anywhere even when it's not needed. I've deployed a fix, and now everything seems to be running smoothly again:

Now the server has reduced its workload by 1000+ requests per minute, and the freed resources can now be used by others.
 
```
$ grep -oP 'code=\KH\d+' /tmp/oneminute2 | sort -n | uniq -c
     69 H27
$ grep -oP 'status=\K\d+' /tmp/oneminute2 | sort -n | uniq -c
   9322 200
   1019 301 <--- This is because of my last fix.
      1 304
     50 400
   3220 403
     29 404
     69 499
```
